autometa/config/databases.py

```diff
@@ -66,230 +97,504 @@ class Databases:
+            Will update provided `options` in `self.config`.
 
         Raises
         -------
         ConnectionError
             NCBI file download failed.
 
         """
-        # Download required NCBI database files
-        if not os.path.exists(self.ncbi_dir):
-            os.makedirs(self.ncbi_dir)
-        host = DEFAULT_CONFIG.get('ncbi', 'host')
         for option in options:
-            ftp_fullpath = DEFAULT_CONFIG.get('database_urls', option)
+            host = self.config.get('ncbi', 'host')
+            ftp_fullpath = self.config.get('database_urls', option)
             ftp_fpath = ftp_fullpath.split(host)[-1]
-            if self.config.has_option('ncbi', option):
+
+            if self.config.has_option('ncbi', option) and self.config.get('ncbi', option) is not None:
                 outfpath = self.config.get('ncbi', option)
             else:
                 outfname = os.path.basename(ftp_fpath)
                 outfpath = os.path.join(self.ncbi_dir, outfname)
-            if not self.dryrun and not os.path.exists(outfpath):
-                with FTP(host) as ftp, open(outfpath, 'wb') as fp:
-                    ftp.login()
-                    logger.debug(f'starting {option} download')
-                    result = ftp.retrbinary(f'RETR {ftp_fpath}', fp.write)
-                    if not result.startswith('226 Transfer complete'):
-                        raise ConnectionError(f'{option} download failed')
-                    ftp.quit()
+
             logger.debug(f'UPDATE: (ncbi,{option}): {outfpath}')
             self.config.set('ncbi', option, outfpath)
-        # Extract/format respective NCBI files
-        self.extract_taxdump()
-        self.format_nr()
 
-    def update_markers(self, options):
-        """Update single-copy markers hmms and cutoffs.
+            if self.dryrun:
+                return
+
+            with FTP(host) as ftp, open(outfpath, 'wb') as fp:
+                ftp.login()
+                logger.debug(f'starting {option} download')
+                result = ftp.retrbinary(f'RETR {ftp_fpath}', fp.write)
+                if not result.startswith('226 Transfer complete'):
+                    raise ConnectionError(f'{option} download failed')
+                ftp.quit()
+        if 'taxdump' in options:
+            self.extract_taxdump()
+        if 'nr' in options:
+            self.format_nr()
+
+    def press_hmms(self):
+        """hmmpress markers database files.
+
+        Returns
+        -------
+        NoneType
+
+        """
+        hmms = (os.path.join(self.markers_dir, fn) for fn in os.listdir(self.markers_dir) if fn.endswith('.hmm'))
+        hmm_search_str = os.path.join('autometa/databases/markers/','*.h3*')
+        pressed_hmms = glob(hmm_search_str)
+        pressed_hmms = set(os.path.realpath(os.path.splitext(fp)[0]) for fp in pressed_hmms)
+        for fp in hmms:
+            if fp in pressed_hmms:
+                continue
+            hmmer.hmmpress(fp)
```

Comment:
Here it would make sense to write checksum files for the pressed databases when the command has finished successfully (or within hmmer.py, I guess). When these files are to be used, then we can check the checksum.

Reply:
Do you mean write a checksum for each of the four files that are created by `hmmpress`? 

e.g. 

1. `markers.hmm.h3p`
2. `markers.hmm.h3i`
3. `markers.hmm.h3m`
4. `markers.hmm.h3f`

Reply:
Yes. If we have a convention that we always make a checksum (either as a separate file or in some special checksums file) after a file is created, then the pipeline can always check whether a checksum is available for a particular file, and we can be confident that the file is good if it's current checksum (i.e. recalculated) matches the one we have.

Reply:
This is being done now in `press_hmms`